lib/auth/clusterconfig/clusterconfigv1/service.go

```diff
@@ -258,3 +263,180 @@ func (s Service) ResetAuthPreference(ctx context.Context, req *clusterconfigpb.R
 
 	return nil, trace.LimitExceeded("failed to reset networking config within %v iterations", iterationLimit)
 }
+
+// GetClusterNetworkingConfig returns the locally cached networking configuration.
+func (s *Service) GetClusterNetworkingConfig(ctx context.Context, _ *clusterconfigpb.GetClusterNetworkingConfigRequest) (*types.ClusterNetworkingConfigV2, error) {
+	authzCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := authzCtx.CheckAccessToKind(types.KindClusterNetworkingConfig, types.VerbRead); err != nil {
+		if err2 := authzCtx.CheckAccessToKind(types.KindClusterConfig, types.VerbRead); err2 != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
```

Comment:
What's going to happen when the outer `if` returns error, but the inner one not? And why, in general, we need two checks?